sig-scheduling/CONTRIBUTING.md

```diff
@@ -72,6 +72,26 @@ to the issue/PR, which helps to preserve the context.
 
 * Always open an issue for a TODO or a follow-up just in case you forget it.
 
+### Technical and style guidelines
+
+The following guidelines apply primarily to kube-scheduler, but some subprojects
+might also adhere to them.
+
+- Follow [effective go](https://go.dev/doc/effective_go) guidelines.
+- Naming:
+  - Length: As a rule-of-thumb, the length of a variable name should be
+    proportional to the size of the scope where it is used and inversely
+    proportional to the number of times that it is used.
+- In tests:
+  - Follow a [DAMP principle](https://stackoverflow.com/a/11837973).
+  - Use `cmp.Diff` instead of `reflect.Equal`, to provide useful comparisons.
+  - Avoid creating or using assertion libraries.
+    Use standard `t.Error` or `t.Fatal`, as necessary.
+
```

Comment:
Add a bullet (or a separate section) to hold high-level guidelines:

- think twice about consumers beyond default-scheduler: like cluster-autoscaler and out-of-tree plugins
- (and I don't think we document some nuances of each extension points very well. Some high-level guidances would be a good start)
- ...

Reply:
The nuances should be in the `devel` folder, not here.